src/components/flyout/deviceDeletionFlyout.js

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import * as actions from '../../actions';
+import Spinner from '../spinner/spinner';
+import Delete from '../../assets/icons/Delete_dark.svg';
+import CancelX from '../../assets/icons/CancelX.svg';
+import TooltipSvg from '../../assets/icons/Tooltip.svg';
+import ApiService from '../../common/apiService';
+import lang from '../../common/lang';
+import PcsBtn from '../shared/pcsBtn/pcsBtn';
+import SummarySection from '../shared/summarySection/summarySection';
+
+import './deviceDeletionFlyout.css';
+
+class DeviceDeletionFlyout extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      showSpinner: false,
+      deviceDeleted: false
+    };
+
+    this.onConfirm = this.onConfirm.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { devices } = nextProps;
+    if (devices.length !== this.props.devices.length) {
+      this.setState({
+        showSpinner: false,
+        deviceDeleted: false
+      });
+    }
+  }
+
+  onConfirm(physicalDevices) {
+    this.setState({ showSpinner: true });
+    Promise.all(
+      physicalDevices.map(({ Id }) => ApiService.deleteDevice(Id))
+    ).then((res) => {
+      this.props.actions.updateDevicesList(physicalDevices);
+      this.setState({
+        showSpinner: false,
+        deviceDeleted: true
+      });
+    });
+  }
+
+  render() {
+    const { devices } = this.props;
+    const physicalDevices = devices.filter(({ IsSimulated }) => !IsSimulated);
+    const IsSimulated = devices.some(({ IsSimulated }) => IsSimulated);
```

Comment:
No need to loop the devices list again, just check the lengths.

`const IsSimulated = devices.length !== physicalDevices.length;`